src/main/java/org/tikv/common/log/SlowLogImpl.java

```diff
@@ -68,45 +64,71 @@ public synchronized SlowLogSpan start(String name) {
     return slowLogSpan;
   }
 
+  @Override
+  public long getTraceId() {
+    return traceId;
+  }
+
+  @Override
+  public long getThresholdMS() {
+    return slowThresholdMS;
+  }
+
   @Override
   public void setError(Throwable err) {
     this.error = err;
   }
 
   @Override
   public void log() {
+    if (error != null || timeExceeded()) {
+      logger.warn("SlowLog:" + getSlowLogJson());
+    }
+  }
+
+  boolean timeExceeded() {
     long currentNS = System.nanoTime();
-    long currentMS = startMS + (currentNS - startNS) / 1_000_000;
-    if (error != null || (slowThresholdMS >= 0 && currentMS - startMS > slowThresholdMS)) {
-      logger.warn("SlowLog:" + getSlowLogString(currentMS));
+    long durationMS = (currentNS - startNS) / 1_000_000;
+    if (slowThresholdMS >= 0 && durationMS > slowThresholdMS) {
+      SimpleDateFormat dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
+      logger.warn(
+          String.format(
+              "A slow request spent %s ms. start=%s, end=%s",
+              durationMS, dateFormat.format(startMS), dateFormat.format(startMS + durationMS)));
+      return true;
     }
+    return false;
   }
 
-  private String getSlowLogString(long currentMS) {
-    SimpleDateFormat dateFormat = getSimpleDateFormat();
+  JsonObject getSlowLogJson() {
     JsonObject jsonObject = new JsonObject();
 
-    jsonObject.addProperty("start", dateFormat.format(startMS));
-    jsonObject.addProperty("end", dateFormat.format(currentMS));
-    jsonObject.addProperty("duration", (currentMS - startMS) + "ms");
     if (error != null) {
       jsonObject.addProperty("error", error.getMessage());
     }
 
-    for (Map.Entry<String, String> entry : properties.entrySet()) {
-      jsonObject.addProperty(entry.getKey(), entry.getValue());
-    }
+    jsonObject.addProperty("trace_id", toUnsignedBigInteger(traceId));
```

Comment:
`trace_id` should come first before `error`

Reply:
Fixed.